pkg/storage/types/dv.go

```diff
@@ -111,22 +112,15 @@ func GenerateDataVolumeFromTemplate(clientset kubecli.KubevirtClient, dataVolume
 	newDataVolume := &cdiv1.DataVolume{}
 	newDataVolume.Spec = *dataVolumeTemplate.Spec.DeepCopy()
 	newDataVolume.ObjectMeta = *dataVolumeTemplate.ObjectMeta.DeepCopy()
-
-	labels := map[string]string{}
-	for k, v := range dataVolumeTemplate.Labels {
-		labels[k] = v
+	newDataVolume.ObjectMeta.Labels = maps.Clone(dataVolumeTemplate.Labels)
+	if newDataVolume.ObjectMeta.Labels == nil {
+		newDataVolume.ObjectMeta.Labels = make(map[string]string)
```

Comment:
@iholder101 WDYT about this (not sure it's better, but I think it's maybe more readable)?
```suggestion

	if dataVolumeTemplate.Labels == nil {
		newDataVolume.ObjectMeta.Labels = make(map[string]string)
	} else {
		newDataVolume.ObjectMeta.Labels = maps.Clone(dataVolumeTemplate.Labels)
	}	
```